pkg/envoy/proxy_test.go

```diff
@@ -59,23 +83,27 @@ func TestProxy(t *testing.T) {
 		return true
 	}, 2*time.Second, 50*time.Millisecond)
 
+	// Check the output from fake-envoy to make sure we didn't hit an error case
+	require.Empty(t, envoyOut.String())
+	require.Empty(t, envoyErr.String())
+
 	// Check that fake-envoy was able to read the config from the pipe.
-	require.Equal(t, bootstrapConfig, output.ConfigData)
+	assert.Equal(t, string(bootstrapConfig), string(output.ConfigData))
 
 	// Check that we're correctly configuring the log level.
-	require.Contains(t, string(output.Args), "--log-level warn")
+	assert.Contains(t, string(output.Args), "--log-level warn")
 
 	// Check that we're disabling hot restarts.
-	require.Contains(t, string(output.Args), "--disable-hot-restart")
+	assert.Contains(t, string(output.Args), "--disable-hot-restart")
 
 	// Check the process is still running.
-	require.NoError(t, p.cmd.Process.Signal(syscall.Signal(0)))
+	assert.NoError(t, p.cmd.Process.Signal(syscall.Signal(0)))
 
 	// Ensure Kill kills and reaps the process.
 	require.NoError(t, p.Kill())
-
 	require.Eventually(t, func() bool {
-		return p.cmd.Process.Signal(syscall.Signal(0)) == os.ErrProcessDone
+		err := p.cmd.Process.Signal(syscall.Signal(0))
+		return errors.Is(err, os.ErrProcessDone)
```

Comment:
This was something the linter suggested as an improvement in order to support wrapped errors